woocommerce/woocommerce-ios
# Xcode 14.3.1 support

## Description
This PR rebases #9356 on top of trunk, adds the new Xcode `14.3.1` image configuration on the CI pipelines, adjust some target configurations and fixes a couple of minor warnings (see commit breakdown).

Important to mention that the `ENABLE_MODULE_VERIFIER` added by Xcode 14.3 ended up failing for the targets `UITestsFoundation` and `Hardware`, so this PR also removes this configuration (and related ones).

## Testing instructions
Run in Xcode 14.3.1 and ensure everything works fine. Wait for green CI results.

---
